ArchiSteamFarm/Steam/Interaction/Commands.cs

```diff
@@ -3257,6 +3427,22 @@ internal void OnNewLicenseList() {
 		return responses.Count > 0 ? string.Join(Environment.NewLine, responses) : null;
 	}
 
+	private static HashSet<EAssetRarity>? ParseAssetRarities(string assetRaritiesText) {
+		string[] assetRaritiesArgs = assetRaritiesText.Split(SharedInfo.ListElementSeparators, StringSplitOptions.RemoveEmptyEntries);
+
+		HashSet<EAssetRarity> assetRarities = [];
```

Comment:
We can optimize memory allocation by using the constructor here with an explicit capacity:
```suggestion
		HashSet<EAssetRarity> assetRarities = new(assetRaritiesArgs.Length);
```

Reply:
We can't because user could provide 300 same entires for which we'll allocate (unnecessarily) 300 entries.

It's better the way it is.

Reply:
You are right. Maybe `Math.Max(assetRaritiesArgs.Length, Enum.GetValues<ETest>().Length)`? My guess is that either compiler or JIT would take care of that constant. 🤔

Reply:
You're adding extra overhead to skip *maybe*, *very unlikely* case of extending the underlying array for extra entries.

When I was analyzing it last time I came to simple conclusion: either we know in advance what the value is, we don't need to calculate it and we're always sure the resulting set will be AT LEAST the value specified, or we don't specify one at all. Specifying value too large (like your suggestion) is not only extra overhead but also extra allocation if not used. I can still duplicate 300 same rarities and you'll allocate 20 or whatever rarities we have defined instead of default which is I think 7 or so.

Reply:
Also it's not const, it'd be static readonly at best, JIT wouldn't do any magic here, you'd need to allocate array for enum values to get their length, and then on each call taking bigger value out of precalculated and provided. You lose hundreds more CPU cycles for that in comparison with hitting unlikely case of underlying array extension.

This is like primary example of premature optimization that hurts more than it helps.

Reply:
As a side note it could be possible to precalculate it using source generator, but you'd still have a possibility of allocating more than you need, and you'd add like a hundred extra lines of code to avoid unlikely case of extending the underlying array :joy: